### ISSUE TITLE
How to localize the Kano UI?

### ISSUE THREAD

**Issue by quantissimo** [KanoComputing/kano-desktop#113]
Hello, I'd like to localize Kano UI into another language for my kid. Can you point me how to do so - which directories contain the UI strings? Where to put the translated locale strings?

Other people also want to contribute with localizations into other languages: http://world.kano.me/forum/topic/how-to-localize-the-kano-ui

**Comment by emnullfuenf** [comment]
I am trying to help translating since more than a year. Team Kano answered me in December 2014 and May 2015 that it should be possible soon.
Last time i had a look in the code i saw that a lot of Strings were hard coded.

**Comment by quantissimo** [comment]
[sign] So, for 2 years now, there's no progress in the localization field... It seems they target English-speaking kids only. It's a pity

**Comment by alex5imon** [comment]
@quantissimo @emnullfuenf I understand your frustrations. We haven't had the time yet to work on the infrastructure for the i18n to happened, we are a small development team. I'm pushing hard on this, but it's a business decision.

**Comment by hzulla** [comment]
Yes, it's somewhat frustrating. Kano OS is promising, but we have decided to settle for a vanilla Raspbian install, instead.

**Comment by alex5imon** [comment]
@quantissimo @hzulla @emnullfuenf  we are working on new content and interfaces for Kano OS. Both take into account localisation. We will start opening the text to community translators as soon as we release the first version (end of April 2016)

**Comment by hzulla** [comment]
Good luck.

**Comment by quantissimo** [comment]
Good news. Please update us here about its progress

**Comment by g-guth** [comment]
Any news on this topic?

**Comment by alex5imon** [comment]
@hzulla you are not mistaken, but I'm sure someone @TeamKano will be able to help @radujipa blink blink

**Comment by hzulla** [comment]
@alex5imon Good luck for your new endeavours!

**Comment by alex5imon** [comment]
thanks @hzulla !

**Comment by equipe23** [comment]
@hzulla it seems you looked into this for other languages: do you know how I can get/build a French localized version of Kano?

**Comment by hzulla** [comment]
@equipe23 No. I haven't looked into Kano i18n since some time. You need to ask the Kano team instead.

**Comment by equipe23** [comment]
Thanks @hzulla.
@TeamKano is there a way to get/build a French localized version of Kano?

**Comment by leoncvlt** [comment]
@TeamKano any news on this? I am sure there are plenty of volunteers willing to help bringing Kano to non-english speaking kids, me included (as a volunteer, not as a kid 😆)

**Comment by equipe23** [comment]
+1 never had an answer.
@TeamKano ?

**Comment by Ealdwulf** [comment]
Sorry for the slow response - the code part is largely internationalised now, but we can't make any promised yet as to when support for particular languages will be released. Thanks for your interest.

**Comment by Assassyn** [comment]
It is a great news. Question now is how to help you with translating to another language?

**Comment by didipluto** [comment]
@Ealdwulf @TeamKano I am from China and I want to be a volunteer to help with translating kano to Chinese.

**Comment by wooyek** [comment]
I have take the liberty and added [transifex support to terminal-quest](https://github.com/KanoComputing/terminal-quest/pull/98). 

@KanoComputing can you pinpoint which projects need such contributions to enable i18n support thus opening them to integration with translation tools? 

I think I speak for mamy people here, we want to help and we'll jump to the opportunity to translate what will land on https://www.transifex.com/kano/

**Comment by k-berger** [comment]
Hey guys! Looking for a way to help @TeamKano with i8n, too. If there is anything left to be done on the infrastructural part, document that as issues tagged free for grabs. So any interested contributor can branch and PR stuff. 
Also, I'd be happy to help with localizations for german and french. My son speaks those two languages, but his english skills are not yet good enough to fully use the potential of Kano.
Let me know what I can do - I'm pretty sure there are other I-am-a-dev-parents and others who will be volunteering. We love your product!

**Comment by carloseduleal** [comment]
+1 looking for a way to help @TeamKano with i8n. I would like to be a volunteer to help translating Kano to Portuguese.

**Comment by G3rb** [comment]
+ 1 to help for french translations.

**Comment by mandre** [comment]
Hi folks, there seem to be a lot of people interested in translating Kano into their language and it's truly a pity the developers do not respond to this thread.

Anyway, I took the liberty to start translation projects on Zanata for:
- [Make Snake](https://translate.zanata.org/project/view/make-snake/) - the snake game
- [Terminal Quest](https://translate.zanata.org/project/view/terminal-quest/) - teach the use of the terminal
- [Kano Overworld](https://translate.zanata.org/project/view/kano-overworld/) - the application behind the Story Mode

And wrote some instruction on how you can update your kano distro to take into account the translated files: https://blog.mandre.org/2019/05/05/bringing-kano-to-non-english-speakers.html

**Comment by G3rb** [comment]
Hi. I started yesterday a french translation (based on your Terminal Quest version on Zanata). But when I want to use it I have several problems. And when I compare your po file to the original en_QQ.po or es_AR.po I don't have the same line for the same strings (in files ex : Beep Beep ... is in the file #: ../linux_story/story/challenges/challenge_1.py:23 on your fr.po but on the original the same string is on the #: ../linux_story/story/challenges/challenge_1.py:32) so It can't work. Can you write me which version did you use ? (I tried the 4.0.0 but that's the same). 
Did you get it working ?

**Comment by G3rb** [comment]
python-snake in french works like a charm !

**Comment by mandre** [comment]
> Did you get it working ?

To be honest, I had not tried further than the introduction screen. I gave it a try earlier and indeed it breaks on the first challenge because the translation file contains UTF8 characters and the function to draw a box didn't know how to handle it. I've submitted a patch to terminal-quest to fix the issue:

https://github.com/KanoComputing/terminal-quest/pull/105

With that, I think I should give a quick update on my progress. I've created a set of scripts to upload translation templates to Zanata and pull the latest translations into Kano. There is now translation projects for:

* [kano-apps](https://translate.zanata.org/project/view/kano-apps/) - Launcher for different utilities ([code repository](https://github.com/KanoComputing/kano-apps))
* [kano-dashboard](https://translate.zanata.org/project/view/kano-dashboard/)
* [kano-feedback](https://translate.zanata.org/project/view/kano-feedback/) - Tool to report feedback and bugs ([code repository](https://github.com/KanoComputing/kano-feedback))
* [kano-greeter](https://translate.zanata.org/project/view/kano-greeter/) - Python greeter for Kano OS ([code repository](https://github.com/KanoComputing/kano-greeter))
* [kano-init](https://translate.zanata.org/project/view/kano-init/) - Initialize a fresh installation of the Kanux OS on a Kano kit ([code repository](https://github.com/KanoComputing/kano-init))
* [kano-overworld](https://translate.zanata.org/project/view/kano-overworld/)
* [kano-peripherals](https://translate.zanata.org/project/view/kano-peripherals/) - Support code for peripherals ([code repository](https://github.com/KanoComputing/kano-peripherals))
* [kano-profile](https://translate.zanata.org/project/view/kano-profile/) - Tool to communicate with Kano-World ([code repository](https://github.com/KanoComputing/kano-profile))
* [kano-settings](https://translate.zanata.org/project/view/kano-settings/) - Graphic tool to setup Kanux ([code repository](https://github.com/KanoComputing/kano-settings))
* [kano-toolset](https://translate.zanata.org/project/view/kano-toolset/) - Shared toolset for OS appsd ([code repository](https://github.com/KanoComputing/kano-toolset))
* [kano-updater](https://translate.zanata.org/project/view/kano-updater/) - Update Kano Linux ([code repository](https://github.com/KanoComputing/kano-updater))
* [kano-vnc](https://translate.zanata.org/project/view/kano-vnc/) - VNC wrapper for Kano OS ([code repository](https://github.com/KanoComputing/kano-vnc))
* [kano-widgets](https://translate.zanata.org/project/view/kano-widgets/) - Kano LXPanel widgets ([code repository](https://github.com/KanoComputing/kano-widgets))
* [make-art](https://translate.zanata.org/project/view/make-art/) - App to learn programming using a basic CoffeeScript drawing API ([code repository](https://github.com/KanoComputing/make-art))
* [make-snake](https://translate.zanata.org/project/view/make-snake/) - Snake game running on the terminal ([code repository](https://github.com/KanoComputing/make-snake))
* [terminal-quest](https://translate.zanata.org/project/view/terminal-quest/) - Introduction to terminal commands in the style of a text adventure game ([code repository](https://github.com/KanoComputing/terminal-quest))

Check out the code at https://github.com/mandre/kano-i18n-sync. Time to start contributing translations ;-)

**Comment by Freedos** [comment]
I've created an Zanata account but everything is read only. How ti contribute?

**Comment by G3rb** [comment]
Yes !! It Works !! Thank you for your work, I can now continue the translation.

**Comment by moorhexa** [comment]
@Freedos You have to join a language team first, as described here: http://docs.zanata.org/en/release/user-guide/languages/language-team/

**Comment by TYeniyayla** [comment]
Hello, I think I can help you with the translation into Turkish, how should I continue? Any progress for localization in 2021? 

Now I've started translating via Zanata. But does it have an effect? Am I doing it right? Will it be valid when I finish it?
